lib/bitcoin/wallet/utxo_handler.rb

```diff
@@ -0,0 +1,55 @@
+module Bitcoin
+  module Wallet
+    class UtxoHandler
+      attr_reader :watchings, :spv, :utxo_db, :pendings
+
+      # @param spv [Bitcoin::Node::SPV]
+      # @utxo_db [Bitcoin::Store::UtxoDb]
+      def initialize(spv, utxo_db)
+        @spv = spv
+        @spv.add_observer(self)
+        @utxo_db = utxo_db
+        @logger = Bitcoin::Logger.create(:debug)
+      end
+
+      def update(event, data)
+        @logger.debug "UtxoHandler#update: #{event}, #{data}"
+        send(event, data)
+      end
+
+      private
+
+      # Called when receiving `tx` message from other node.
+      #
+      # - Store utxo if the output of received tx is in the watch list
+      # - Delete from utxo_db for spent output.
+      def tx(data)
+        tx = data.tx
+        tx.outputs.each_with_index do |output, index|
+          next unless watching?(output)
+          out_point = Bitcoin::OutPoint.new(tx.tx_hash, index)
+          utxo_db.save_utxo(out_point, output.value, output.script_pubkey)
+        end
+
+        tx.inputs.each do |input|
+          utxo_db.delete_utxo(input.out_point)
+        end
+      end
+
+      def header(data)
+      end
+
+      def merkleblock(data)
```

Comment:
There doesn't seem to be any implementation, is this one left empty?

Reply:
I add implementation for merkleblock and header message.
When merkleblock is called, spv nodes will store the tx position data into the leveldb.

Reply:
I add implementation for merkleblock and header message.
When merkleblock is called, spv nodes will store the tx position data into the leveldb.